### PR TITLE
Fix corpus count methods

### PR DESCRIPTION
**Pull request by ddaspit**
- add support multiple columns in TextFileText.Count method
- remove MissingRowsAllowed property
- use standard Count implementation for ParallelTextCorpus

### CODE REVIEW

**Comment by johnml1135** [comment]
*[`src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs` line 38 at r1](https://reviewable.io/reviews/sillsdev/machine/121#-NhWNbNIA1JoLyej74o3:-NhWNbNJCqyb_rrDSzX2:b-quu7yg) ([raw file](https://github.com/sillsdev/machine/blob/796e1e4f6b0e9bd1599c2d858f6598982266945d/src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs#L38)):*
> ```Smalltalk
>                         int index = line.IndexOf("\t");
>                         if (index >= 0)
>                             line = line.Substring(index + 1);
> ```

As per the swagger docs: "If a line contains a tab, characters before the tab are used as a unique identifier for the line, characters after the tab are understood as the content of the verse, and if there is another tab following the verse content, characters after this second tab are assumed to be column codes like "ss" etc. for sectioning and other formatting."  
  In other words, we need not just everything after the first tab, but whatever is between the first and second tab (if there is one).

**Comment by johnml1135** [comment]
*[`src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs` line 38 at r1](https://reviewable.io/reviews/sillsdev/machine/121#-NhWNbNIA1JoLyej74o3:-NhWOIp15CJ2EqDpNIcO:b-jpasiv) ([raw file](https://github.com/sillsdev/machine/blob/796e1e4f6b0e9bd1599c2d858f6598982266945d/src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs#L38)):*
<details><summary><i>Previously, johnml1135 (John Lambert) wrote…</i></summary><blockquote>

As per the swagger docs: "If a line contains a tab, characters before the tab are used as a unique identifier for the line, characters after the tab are understood as the content of the verse, and if there is another tab following the verse content, characters after this second tab are assumed to be column codes like "ss" etc. for sectioning and other formatting."  
  In other words, we need not just everything after the first tab, but whatever is between the first and second tab (if there is one).
</blockquote></details>

One way to resolve this is to convert the line to a string and use the same string operations as is `TextFileText.cs`.

**Comment by johnml1135** [comment]
Also default to the most recent docker image for development.  Nice.

**Review by johnml1135 — CHANGES_REQUESTED** [review]
> Reviewed 17 of 17 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/machine/121)* status: all files reviewed, 2 unresolved discussions (waiting on @ddaspit)

**Review by ddaspit — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/machine/121)* status: all files reviewed, 2 unresolved discussions (waiting on @johnml1135)

___
*[`src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs` line 38 at r1](https://reviewable.io/reviews/sillsdev/machine/121#-NhWNbNIA1JoLyej74o3:-NhXt14i5Sz8fUIISltj:b-wpvesl) ([raw file](https://github.com/sillsdev/machine/blob/796e1e4f6b0e9bd1599c2d858f6598982266945d/src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs#L38)):*
<details><summary><i>Previously, johnml1135 (John Lambert) wrote…</i></summary><blockquote>

One way to resolve this is to convert the line to a string and use the same string operations as is `TextFileText.cs`.
</blockquote></details>

This is for a word alignment corpus not a text corpus, so the Swagger docs don't apply.

**Comment by johnml1135** [comment]
*[`src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs` line 38 at r1](https://reviewable.io/reviews/sillsdev/machine/121#-NhWNbNIA1JoLyej74o3:-NhY6rfc-NFKLQyY7KI9:bsrk1gn) ([raw file](https://github.com/sillsdev/machine/blob/796e1e4f6b0e9bd1599c2d858f6598982266945d/src/SIL.Machine/Corpora/TextFileAlignmentCollection.cs#L38)):*
<details><summary><i>Previously, ddaspit (Damien Daspit) wrote…</i></summary><blockquote>

This is for a word alignment corpus not a text corpus, so the Swagger docs don't apply.
</blockquote></details>

ok - I understand.

**Review by johnml1135 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/machine/121)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ddaspit)
